src/algorithm/crba.hxx

```diff
@@ -161,27 +161,41 @@ namespace se3
                      const se3::Model & model,
                      se3::Data & data)
     {
-      typedef typename SizeDepType<JointModel::NV>::template ColsReturn<Matrix6x>::Type ColsBlock;
+      typedef typename SizeDepType<JointModel::NV>::template ColsReturn<Data::Matrix6x>::Type ColsBlock;
       
       const Model::JointIndex & i = (Model::JointIndex) jmodel.id();
       const Model::Index & parent = model.parents[i];
       
       data.Ycrb[parent] += data.liMi[i].act(data.Ycrb[i]);
       
       jdata.U() = data.Ycrb[i] * jdata.S();
-      if (JointModel::NV == -1)
-      {
-        ColsBlock jF
-        = data.Ag.middleCols(jmodel.idx_v(), jmodel.nv());
-      }
-      else
-      {
-        ColsBlock jF
+
+      ColsBlock jF
         = data.Ag.middleCols <JointModel::NV> (jmodel.idx_v());
-      }
+
       forceSet::se3Action(data.oMi[i],jdata.U(),jF);
     }
     
+    static void algo(const se3::JointModelBase<JointModelComposite> & jmodel,
+                     se3::JointDataBase<JointDataComposite> & jdata,
+                     const se3::Model & model,
+                     se3::Data & data)
+    {
+      typedef SizeDepType<JointModel::NV>::ColsReturn<Data::Matrix6x>::Type ColsBlock;
+      
+      const Model::JointIndex & i = (Model::JointIndex) jmodel.id();
+      const Model::Index & parent = model.parents[i];
+      
+      data.Ycrb[parent] += data.liMi[i].act(data.Ycrb[i]);
+      
+      jdata.U() = data.Ycrb[i] * jdata.S();
+      
+      ColsBlock jF
```

Comment:
Please see commit 3af876e and the call to `middleCols` 
According to http://eigen.tuxfamily.org/dox/classEigen_1_1DenseBase.html#aa257ee10f662780e48cd9cec53bbbb86 it seems legit and the trade-off seems acceptable (for crba i have the following bench: from 4,35 us 90% of time & 10% : 4.5 us  to 4.4 us 60% of time, and 4.7 us otherwise )